addrmgr/addrmanager.go

```diff
@@ -1181,13 +1181,15 @@ func (a *AddrManager) GetBestLocalAddress(remoteAddr *wire.NetAddress) *wire.Net
 	return bestAddress
 }
 
-// IsPeerNaValid asserts if the provided local address is routable
-// and reachable from the peer that suggested it.
-func (a *AddrManager) IsPeerNaValid(localAddr, remoteAddr *wire.NetAddress) bool {
+// ValidatePeerNa returns the validity and reachability of the
+// provided local address based on its routablility and reachability
+// from the peer that suggested it.
+func (a *AddrManager) ValidatePeerNa(localAddr, remoteAddr *wire.NetAddress) (bool, int) {
```

Comment:
This changes the public API which would require a major bump version bump.  I'd suggest just introducing the new function, redefining `IsPeerNaValid`, in terms of it and deprecating it.